include/s3select_oper.h

```diff
@@ -2534,7 +2538,12 @@ class base_like : public base_function
     value like_expr_val;
     value escape_expr_val;
     bool constant_state = false;
-    std::unique_ptr<RE2> compiled_regex;
+    #if REGEX_HS
+      hs_database_t* compiled_regex;
```

Comment:
since the semantic is the same and the difference is type (operation on type)
maybe template could be more appropriate